src/json.rs

```diff
@@ -142,11 +142,14 @@ fn json_content_type<B>(req: &RequestParts<B>) -> Result<bool, HeadersAlreadyExt
         return Ok(false);
     };
 
-    Ok(mime.subtype() == "json"
-        || mime
-            .suffix()
-            .filter(|name| name.as_str() == "json")
-            .is_some())
+    let is_json_content_type = mime.type_() == "application"
+        && (mime.subtype() == "json"
+            || mime
+                .suffix()
+                .filter(|name| name.as_str() == "json")
```

Comment:
`Name` implements `PartialEq<&str>` so this `as_str()` shouldn't be needed. Though maybe it would have to be `*name` otherwise? Anyways, was just curious why `.as_str()` is needed, it's readable either way.